rosbag2_storage_default_plugins/src/rosbag2_storage_default_plugins/sqlite/sqlite_wrapper.cpp

```diff
@@ -56,8 +58,33 @@ SqliteWrapper::SqliteWrapper(
         rc << "): " << sqlite3_errstr(rc);
       throw SqliteException{errmsg.str()};
     }
-    prepare_statement("PRAGMA journal_mode = WAL;")->execute_and_reset();
-    prepare_statement("PRAGMA synchronous = NORMAL;")->execute_and_reset();
+
+    auto journal_mode_set = std::find_if(
+      pragmas.begin(), pragmas.end(),
+      [](const auto & pragma) {
+        if (pragma.rfind("journal_mode", 0) == 0) {
```

Comment:
maybe
```suggestion
        return (pragma.rfind("journal_mode", 0) == 0);
```

same below - personally find it more readable